app/models/person.rb

```diff
@@ -54,6 +68,33 @@ def offer_tag_list
     offers.any? ? offers&.map(&:all_tags_unique) : []
   end
 
+  def anonymized_name_and_email
+    "#{anonymize_name} #{anonymize_email}"
+  end
+
+  def anonymize_email
+    return if email.blank?
+
+    at_index = email.index("@")
+    words = email.split(".")
+    top_level_domain = words.pop
+
+    masked_email = "*" * words.join(".").length
+    masked_email[at_index] = "@"
+    masked_email.concat(".", top_level_domain)
+  end
+  
+  def anonymize_name
+    return if name.blank?
+
+    names = name.split(" ")
+    names.map do |n|
+      initial = n.first
+      n = '*' * n.length
+      initial + n[1..-1]
+    end.join(" ")
+  end
+
```

Comment:
This logic isn't part of the Person object's core responsibility so, adhering to the SPR, i'd love to get them extracted into a utility module.

Maybe something like `app/lib/anonymize.rb`, so that they can be used like this:

```ruby
def anonymized_name_and_email
  "#{Anonymize.name(name)} #{Anonymize.email(email)"
end
```

Note: in a departure from rails norms, i personally don't like using mixins for such cases. Happy to say more about that if anyone's interested :).

This would also make these methods very easy to test.

Reply:
I would definitely like to know more about not using mixins for these scenarios :). Also, I have moved this logic to `Anonymize` class.